Added feedback table

tej99 opened the pull request:
Assumptions made:
- 'survey_id' will never be more than 25 characters long 
- 'period' is year/month which will not be more than 25 characters long

tej99 committed:
Added feedback table

tej99 committed:
Added feedback table

JamesGardiner reviewed (changes requested):
Can you update the CHANGELOG? Also, are 25 chars guaranteed to always be enough?

tej99 committed:
CHANGELOG updated

tej99 committed:
CHANGELOG updated

necrophonic committed:
Merge branch 'master' into feature/feedbackform

warrenbailey commented:
Good point @JamesGardiner I missed that, the SQLAlchemy doesn't restrict the size of the String so I think it's possible that it could exceed 25 chars. (https://github.com/ONSdigital/sdx-store/pull/62/files)

warrenbailey commented on postgres/schema.sql:
```diff
@@ -1,4 +1,5 @@
 CREATE TABLE IF NOT EXISTS responses (tx_id uuid PRIMARY KEY, ts timestamp WITH time zone DEFAULT NOW(), invalid boolean DEFAULT NULL, data jsonb);
+CREATE TABLE IF NOT EXISTS feedback_responses (id SERIAL, ts timestamp WITH time zone DEFAULT NOW(), invalid boolean DEFAULT NULL, data jsonb, survey char(25), period char(25));
```
What's the reason for 25 chars?

tej99 replied:
I'm not sure if you can create a column with data type char without having a size and 25 chars seemed sensible. I can change the SQLAlchemy to restrict the size to 25? Unless there's a better way to do this?

JamesGardiner replied:
Use an `@property` and validate the string length in an `@property.setter` method, raising an exception if necessary

tej99 replied:
SQLAlchemy handles that

necrophonic committed:
Removes redundant services and checkout/install of sdx common

No longer needed as individual services manage their own dependencies on a local level and
keys etc no longer need to be generated here.

tej99 committed:
Merge branch 'feature/feedbackform' of https://github.com/ONSdigital/sdx-compose into feature/feedbackform

tej99 committed:
Changes to sql

tej99 committed:
Delete duplication
